### PR TITLE
Update to electron. try affinity flag

### PR DESCRIPTION
**Pull request by chrisnojima**
This updates electron to 2.0.2 and updates the packager and some packager deps.
This also turns on the affinity flag: https://github.com/electron/electron/pull/11501 which reduces the overhead when we have multiple remote windows going

- [x] window
- [x] linux
- [x] darwin
- [x] update webpack preset env

### CODE REVIEW

File: shared/desktop/package.desktop.js [hunk 1]
```diff
@@ -15,7 +15,12 @@ const desktopPath = (...args) => path.join(__dirname, ...args)
 const copySyncFolder = (src, target, onlyExts) => {
   const srcRoot = desktopPath(src)
   const dstRoot = desktopPath(target)
-  const files = klawSync(srcRoot, {filter: item => onlyExts.includes(path.extname(item.path))})
+  const files = klawSync(srcRoot, {
```
Comment (chrisnojima) [inline shared/desktop/package.desktop.js]: klaw passes the folders to us so if we reject it it won't keep going down

File: shared/desktop/package.desktop.js [hunk 2]
```diff
@@ -31,25 +36,23 @@ const argv = minimist(process.argv.slice(2), {string: ['appVersion']})
...
 const shouldBuildAll = argv.all || false
-// $FlowIssue // flow-typed libdef is pretty weak, thinks this might be a boolean
-const shouldBuildAnArch: ?string = argv.arch
-// $FlowIssue // flow-typed libdef is pretty weak, thinks this might be a boolean
-const appVersion: string = argv.appVersion || '0.0.0'
-// $FlowIssue // flow-typed libdef is pretty weak, thinks this might be a boolean
-const comment: string = argv.comment || ''
-// $FlowIssue // flow-typed libdef is pretty weak, thinks this might be a boolean
-const outDir: string = argv.outDir || ''
+const shouldBuildAnArch: string = (argv.arch: any)
```
Comment (chrisnojima) [inline shared/desktop/package.desktop.js]: lets just cast instead of flowissue

File: shared/desktop/package.desktop.js [hunk 3]
```diff
@@ -31,25 +36,23 @@ const argv = minimist(process.argv.slice(2), {string: ['appVersion']})
...
-const packagerOpts = {
+const packagerOpts: any = {
   appBundleId: 'keybase.Electron',
   appCopyright: appCopyright,
   appVersion: appVersion,
   asar: shouldUseAsar,
-  buildVersion: appVersion + comment,
+  buildVersion: String(appVersion) + String(comment),
   dir: desktopPath('./build'),
+  electronVersion: 0,
```
Comment (chrisnojima) [inline shared/desktop/package.desktop.js]: flow doesn't like us assigning a var later to a key that doesn't exist

File: shared/desktop/package.desktop.js [hunk 4]
```diff
@@ -173,15 +182,16 @@ function pack(plat, arch, cb) {
     }
   }
 
-  packager(opts, cb)
+  return packager(opts)
```
Comment (chrisnojima) [inline shared/desktop/package.desktop.js]: we get deprecation warnings if we don't use the promise version

**Comment by chrisnojima** [comment]
@oconnor663 
@zanderz can you do test builds on windows/linux and test that this works. please ensure the menu widget and trackers work as expected

**Comment by oconnor663** [comment]
@chrisnojima it seems ok to me when I do a local build.

**Comment by chrisnojima** [comment]
has to be what is a production type build and not any kind of hot server etc

**Comment by chrisnojima** [comment]
@zanderz when you have a chance can we test a windows build of this

**Comment by chrisnojima** [comment]
@zanderz lets prioritize this monday

**Comment by zanderz** [comment]
Sorry, forgot to mention that I built and sanity checked this for minute,
got a tracker popup, etc.

**Comment by chrisnojima** [comment]
@keybase/react-hackers can someone review. i'm going to kick admin builds off this commit so we can play with it today

**Comment by chrisnojima** [comment]
@oconnor663 / @zanderz  can you double check notifications work

**Comment by chrisnojima** [comment]
@keybase/react-hackers lets review this and get it in

**Comment by adamjspooner** [comment]
I'm guessing the insecure CSP warning is expected.

**Comment by chrisnojima** [comment]
thats only in dev

**Comment by chrisnojima** [comment]
@keybase/react-hackers bump

**Comment by adamjspooner** [comment]
I ran this branch for most of the morning with no issues, but that was with `hot-server` and `start-hot`. Everything seems fine to me. Is that enough to approve it?

**Comment by chrisnojima** [comment]
Better to make a test build with the bot

**Comment by adamjspooner** [comment]
How do I do that?

**Comment by zanderz** [comment]
Sorry, I have missing notifications with build https://prerelease.keybase.io/windows/Keybase_2.1.0-20180607094851%2B399affe0c6.386.exe, which I think is supposed to contain the windows fix. Still looking.

**Comment by zanderz** [comment]
building a new windows one off this branch now

**Comment by adamjspooner** [comment]
I'm not sure it's relevant, but I get this error when I set an exploding message time: ```Error: ERROR CODE 218 - method 'updateCategory' not found in protocol 'keybase.1.gregor' in method keybase.1.gregor.updateCategory```

**Comment by chrisnojima** [comment]
unrelated, i just merged master back in so that kind of stuff won't show up

**Comment by zanderz** [comment]
Notifications work with https://prerelease.keybase.io/windows/Keybase_2.1.0-20180611142735%2B918d49f71.386.exe, LGTM

**Comment by chrisnojima** [comment]
k i'm merging

**Pull request merged**